### PR TITLE
Changed workflow for setting the masking value in the skull stripping

### PR DESCRIPTION
**Pull request by nicmuenster**
This change implements the the same process as in the defacing step to either use a global custom value for all inputs or the minimum value within the input image as the background value when masking an inage

### CODE REVIEW

**Review by neuronflow — COMMENTED** [review]
Do we need changes here?

https://github.com/BrainLesion/preprocessing/blob/22661a6c78c644f86c379d6219a4348008f76256/brainles_preprocessing/brain_extraction/synthstrip.py#L220

**Comment by nicmuenster** [comment]
From my understanding, updating the synthstrip init is still necessary, given that while the center modality might be handled without the apply_mask function in the preprocessor workflow (as seen here)

`atlas_mask = self.center_modality.extract_brain_region(
            brain_extractor=self.brain_extractor,
            bet_dir_path=bet_dir,
            use_gpu=self.use_gpu,
        )`

the moving modalities are then afterwards still handled by the apply_mask function:

`for moving_modality in self.moving_modalities:
moving_modality.apply_bet_mask(
                brain_extractor=self.brain_extractor,
                mask_path=atlas_mask,
                bet_dir=bet_dir,
            )
`        
One thing to consider however, is that in the synthstrip workflow the minimum between 0 and the input min is taken, so min values above 0 would be handled slightly different now, but this missmatch actually also exists for the current state as well.

**Comment by neuronflow** [comment]
@nicmuenster merge conflict needs to be resolved :)

**Comment by neuronflow** [comment]
@uturkbey do you have time to look into this?

**Comment by neuronflow** [comment]
> ## Pull request overview
> Copilot reviewed 3 out of 3 changed files in this pull request and generated 2 comments.
> 
> Comments suppressed due to low confidence (2)
> **brainles_preprocessing/brain_extraction/synthstrip.py:39**
> 
> * `masking_value` is now accepted and passed into `BrainExtractor`, but SynthStrip still hard-codes the masked background value inside `extract()` (see `bg = np.min([0, img_data.min()])` and `img_data[mask == 0] = bg` at lines 225-226). This means center-modality skull-stripped outputs written by `SynthStripExtractor.extract()` will ignore the new `masking_value` behavior described in the docstring/PR.
> 
> ```
>             border (int): Mask border threshold in mm. Defaults to 1.
>             masking_value (Optional[Union[int, float]], optional): global value to be inserted in the masked areas. Default is None which leads to the minimum of each respective image.
> 
>         """
>         super().__init__(masking_value=masking_value)
> ```
> 
> **brainles_preprocessing/brain_extraction/brain_extractor.py:79**
> 
> * The new `masking_value` behavior in `apply_mask()` (global override vs per-image minimum) is not covered by assertions in the existing tests. Current tests only check that an output file is created, so regressions in the chosen background value or dtype casting would go unnoticed.
> 
> ```
>         # check whether a global masking value was passed, otherwise choose minimum
>         if self.masking_value is None:
>             current_masking_value = np.min(input_data)
>         else:
>             current_masking_value = (
> ```

@nicmuenster please also have a look at the surpressed comments

**Comment by nicmuenster** [comment]
I implmented the first comment, but I am not sure about the second one to be honest. I would also maybe first implement the modality specific masking values before creating specific tests for edge cases

File: brainles_preprocessing/brain_extraction/synthstrip.py [hunk 1]
```diff
@@ -219,8 +222,14 @@ def extract(
...
-        bg = np.min([0, img_data.min()])
-        img_data[mask == 0] = bg
+        # check whether a global masking value was passed, otherwise choose minimum
+        if self.masking_value is None:
+            current_masking_value = np.min(img_data)
+        else:
+            current_masking_value = (
+                np.array(self.masking_value).astype(img_data.dtype).item()
+            )
+        img_data[mask == 0] = current_masking_value
```
[the remark this answers is not in the document]
Comment (neuronflow) [reply brainles_preprocessing/brain_extraction/synthstrip.py:232]: @nicmuenster sounds like a valid comment ;)
